client/components/Auth/Auth.jsx

```diff
@@ -0,0 +1,159 @@
+      .required('E-mail is required'),
+    password: yup
+      .string()
+      .typeError('Password should be a string')
+      .required('Password is required'),
+  };
+  const initialValues = { email: '', password: '' };
+
+  if (isRegister) {
+    validationObject.confirm = yup
+      .string()
+      .oneOf([yup.ref('password')], "Passwords doesn't match")
+      .required('Confirm password is required');
+    initialValues.confirm = '';
+  }
+  const validationsSchema = yup.object().shape(validationObject);
+
+  return (
+    <main className="auth">
+      <h1>{isRegister ? 'Registration' : 'LogIn'}</h1>
+
+      <div className="decoration decoration_light" />
+
+      <Formik
+        initialValues={initialValues}
+        validationSchema={validationsSchema}
+        onSubmit={(values, { setSubmitting }) => {
+          setSubmitting(false);
+          onSubmit(values.email, values.password);
+        }}
+      >
+        {({
+          values,
+          errors,
+          touched,
+          handleChange,
+          handleBlur,
+          handleSubmit,
+          isSubmitting,
+          dirty,
+          isValid,
+        }) => (
+          <form className="auth__form" onSubmit={handleSubmit}>
+            <label htmlFor="email-id" className="row">
+              E-mail
+              <input
+                className={`auth__form__input${
+                  errors.email && touched.email ? ' auth__form__input_wrong' : ''
+                }`}
+                id="email-id"
+                type="email"
+                name="email"
+                onChange={handleChange}
+                onBlur={handleBlur}
+                value={values.email}
+              />
+            </label>
+            <label htmlFor="password-id" className="row">
+              Password
+              <input
+                className={`auth__form__input${
+                  errors.password && touched.password ? ' auth__form__input_wrong' : ''
+                }`}
+                id="password-id"
+                type="password"
+                name="password"
+                onChange={handleChange}
+                onBlur={handleBlur}
+                value={values.password}
+              />
+            </label>
+
+            {isRegister ? (
+              <label htmlFor="confirm-id" className="row">
+                Confirm password
+                <input
+                  className={`auth__form__input${
+                    errors.confirm && touched.confirm ? ' auth__form__input_wrong' : ''
+                  }`}
+                  id="confirm-id"
+                  type="password"
+                  name="confirm"
+                  onChange={handleChange}
+                  onBlur={handleBlur}
+                  value={values.confirm}
+                />
+              </label>
+            ) : (
+              ''
+            )}
+            <p className="auth__error">
+              {(errors.email && touched.email && errors.email) ||
```

Comment:
@IgorChaikin I think `errors.email` appears twice for same `&&` condition. Same for `errors.password` and `errors.confirm`